### ISSUE TITLE
Report useless check when accessing array elements in `no-useless-undefined`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1689]
#### Fail

```js
const foo = index >= 0 ? array[index] : undefined;
const foo = index < 0 ? undefined : array[index];
```

```js
const foo = index <= array.length - 1 ? array[index] : undefined;
const foo = index > array.length - 1 ? undefined : array[index];
```

```js
const foo = index >= 5 ? array[index - 5] : undefined;
```

#### Pass

```js
const foo = array[index];
```

```js
const foo = index >= 0 ? array[index] : bar;
```

I'm not sure if we should add this to  `no-useless-undefined` or add a new rule.

**Comment by sindresorhus** [comment]
I would add it to `no-useless-undefined`.
